File: Golang/ex14.go
package main

import (
	"fmt"
	"sync"
)

type Configuracao struct {
	Parametro string
}

var instancia *Configuracao
var once sync.Once

func GetConfiguracao() *Configuracao {
	once.Do(func() {
		instancia = &Configuracao{Parametro: "Valor padrão"}
	})
	return instancia
}

func main() {
	config1 := GetConfiguracao()
	fmt.Println("Configuração 1:", config1.Parametro)

	config2 := GetConfiguracao()
	config2.Parametro = "Novo valor"

	fmt.Println("Configuração 1 após alteração:", config1.Parametro)
	fmt.Println("Configuração 2:", config2.Parametro)
}

File: python/ex14.py
class Configuracao:
    _instancia = None  

    def __new__(cls):
        if cls._instancia is None:
            cls._instancia = super(Configuracao, cls).__new__(cls)
            
            cls._instancia.configuracoes = {}
        return cls._instancia

    def set_configuracao(self, chave, valor):
        self.configuracoes[chave] = valor

    def get_configuracao(self, chave):
        return self.configuracoes.get(chave, None)


def main():
    
    config1 = Configuracao()
    config2 = Configuracao()
